scripts/stackstorm.js

```diff
@@ -409,32 +419,36 @@ module.exports = function(robot) {
       }
     });
 
-    // Add an interval which tries to re-load the commands
-    commands_load_interval = setInterval(loadCommands.bind(self), (env.ST2_COMMANDS_RELOAD_INTERVAL * 1000));
-
     // Initial command loading
-    loadCommands();
+    loadCommands(env.EXIT_ON_FAILURES);
 
     // Install SIGUSR2 handler which reloads the command
     install_sigusr2_handler();
+
+    // Add an interval which tries to re-load the commands
+    commands_load_interval = setInterval(loadCommands.bind(self, false), (env.ST2_COMMANDS_RELOAD_INTERVAL * 1000));
   }
 
-  function stop() {
+  function stop(exitFromProcess) {
```

Comment:
Since `exitFromProcess` is an options object, we should rename it to something like `opts` or `options`.